src/plugins/intel_cpu/src/nodes/kernels/jit_kernel_base.hpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <math.h>
+#include <dnnl_extension_utils.h>
+#include <cpu/x64/jit_generator.hpp>
+#include <cpu/x64/cpu_isa_traits.hpp>
+#include "jit_base.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu;
+
+class jit_kernel_base : public jit_base {
+public:
+    using jit_base::jit_base;
+
+    void generate() override {
+        this->preamble();
+
+        registers_pool_ = RegistersPool::create(max_cpu_isa_);
+        stack_allocator_ = std::unique_ptr<StackAllocator>(new StackAllocator{*this});
+
+        generate_impl();
+
+        registers_pool_.reset();
+        stack_allocator_.reset();
+
+        this->postamble();
+
+        for(auto& record : emitters_map_) {
+            record.second->emit_data();
+        }
+    }
+
+    virtual void generate_impl() = 0;
+
+    void emu_vgatherdps(const Xbyak::Xmm &xmm_val,
+                        const Xbyak::Reg64 &reg_addr,
+                        const Xbyak::Xmm &xmm_index,
+                        const int &scale,
+                        const int &disp,
+                        const Xbyak::Reg &reg_mask) {
+        const size_t kDataTypeSize = sizeof(float);
+        if (is_valid_isa(x64::avx512_core)) {
+            assert(reg_mask.isOPMASK());
+            vgatherdps(xmm_val, ptr[reg_addr + xmm_index * scale + disp]);
+        } else if (is_valid_isa(x64::avx2)) {
+            assert(reg_mask.isYMM());
+            Xbyak::Ymm ymm_mask{reg_mask.getIdx()};
+            vgatherdps(xmm_val, ptr[reg_addr + xmm_index * scale + disp], ymm_mask);
+        } else {
+            const size_t kSimdWidth = x64::cpu_isa_traits<x64::sse41>::vlen / kDataTypeSize;
+            assert(reg_mask.isXMM());
+            Xbyak::Xmm xmm_mask{reg_mask.getIdx()};
+            assert(xmm_val.getKind() == xmm_index.getKind());
+            assert(xmm_index.getKind() == xmm_mask.getKind());
+
+            std::vector<Xbyak::Reg> not_available_reg{reg_addr};
+            const Xbyak::Reg64 idx = register_pool()->getInplaceFree<Xbyak::Reg64>(not_available_reg);
```

Comment:
The idea is to add factory methods to the `jit_base` level, so that we can hide the whole register management mechanism to decouple user code from the utility entities.

Reply:
What do you mean by `decouple user code from the utility entities`? You will still need to include registers pool header everywhere you are using register_pool(), no matter if it's located in jit_kernel_base or jit_base. Registers pool lifetime is already encapsulated in jit_kernel_base, so what's the point?